File: frontend/lib/vis/test_base64.js


import { isStrictBase64 } from "./utils/base64_validator.js";

export const validateAuth = async (data) => {
    let result = {
        success: false,
        errors: [],
        data: data
    };

    const successData = {};
    let flag = true;
    {
        flag = true;flag = true;

if (data["avatar"] === undefined || data["avatar"] === null) {
    result.errors.push({
        path: "avatar",
        message: "Invalid value."
    });
    flag = false;
} else if (typeof data["avatar"] !== "string") {
    result.errors.push({
        path: "avatar",
        message: "Invalid value."
    });
    flag = false;
} else if (!isStrictBase64(data["avatar"], false)) {
    result.errors.push({
        path: "avatar",
        message: "Invalid value."
    });
    flag = false;
} else {
    const decodedLength = typeof Buffer !== 'undefined' 
        ? Buffer.from(data["avatar"], 'base64').length 
        : atob(data["avatar"].replace(/-/g, '+').replace(/_/g, '/')).length;
    if (decodedLength > 1048576) {
        result.errors.push({
            path: "avatar",
            message: "Invalid value."
        });
        flag = false;
    }
}
if (flag) {
            successData.avatar = data.avatar;
        }
    }
    {
        flag = true;flag = true;

if (data["sessionToken"] === undefined || data["sessionToken"] === null) {
    result.errors.push({
        path: "sessionToken",
        message: "Invalid value."
    });
    flag = false;
} else if (typeof data["sessionToken"] !== "string") {
    result.errors.push({
        path: "sessionToken",
        message: "Invalid value."
    });
    flag = false;
} else if (!isStrictBase64(data["sessionToken"], true)) {
    result.errors.push({
        path: "sessionToken",
        message: "Invalid value."
    });
    flag = false;
} else {
    const decodedLength = typeof Buffer !== 'undefined' 
        ? Buffer.from(data["sessionToken"], 'base64url').length 
        : atob(data["sessionToken"].replace(/-/g, '+').replace(/_/g, '/')).length;
    if (decodedLength < 32) {
        result.errors.push({
            path: "sessionToken",
            message: "Invalid value."
        });
        flag = false;
    }
    if (decodedLength > 64) {
        result.errors.push({
            path: "sessionToken",
            message: "Invalid value."
        });
        flag = false;
    }
    data["sessionToken"] = data["sessionToken"].trim();
}
if (flag) {
            successData.sessionToken = data.sessionToken;
        }
    }
    {
        flag = true;flag = true;

if (data.tags === undefined || data.tags === null) {
    result.errors.push({
        path: "tags",
        message: "Invalid value."
    });
    flag = false;
} else if (!Array.isArray(data.tags)) {
    result.errors.push({
        path: "tags",
        message: "Invalid value."
    });
    flag = false;
} else {

    for (let i = 0; i < data.tags.length; i++) {
        const previousFlag = flag;
        flag = true;

        const itemPath = "tags" + "[" + i + "]";
        data[itemPath] = data.tags[i];flag = true;

if (data[itemPath] === undefined || data[itemPath] === null) {
    result.errors.push({
        path: itemPath,
        message: "Invalid value."
    });
    flag = false;
} else if (typeof data[itemPath] !== "string") {
    result.errors.push({
        path: itemPath,
        message: "Invalid value."
    });
    flag = false;
} else if (!isStrictBase64(data[itemPath], false)) {
    result.errors.push({
        path: itemPath,
        message: "Invalid value."
    });
    flag = false;
} else {
}


        data.tags[i] = data[itemPath];
        delete data[itemPath];

        const itemFlag = flag;
        flag = previousFlag && itemFlag;}
}if (flag) {
            successData.tags = data.tags;
        }
    }

    if (result.errors.length === 0) {
        result = {
            success: true,
            errors: [],
            data: successData
        };
    }

    return result;
};

File: backend/lib/vis/test_base64.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;

using Generated.Validation.Utils;


namespace Generated.Validation
{
    public static class Validators
    {
        public static async Task<ValidationResult> ValidateAuth(Dictionary<string, object?> data)
        {
            var result = new ValidationResult
            {
                Success = false,
                Errors = new List<ValidationError>(),
                Data = data
            };

            var successData = new Dictionary<string, object?>();
            bool flag = true;
            {
                flag = true;bool flag = true;

if (!data.TryGetValue("avatar", out var rawValue) || rawValue == null)
{
    result.Errors.Add(new ValidationError
    {
        Path = "avatar",
        Message = "Invalid value."
    });
    flag = false;
}
else if (rawValue is not string value)
{
    result.Errors.Add(new ValidationError
    {
        Path = "avatar",
        Message = "Invalid value."
    });
    flag = false;
}
else if (!Base64Validator.IsStrictBase64(value, false)) {
    result.Errors.Add(new ValidationError {
        Path = "avatar",
        Message = "Invalid value."
    });
    flag = false;
} else {
    byte[] decodedBytes = Convert.FromBase64String(value.Replace('-', '+').Replace('_', '/'));
    if (decodedBytes.Length > 1048576) {
        result.Errors.Add(new ValidationError {
            Path = "avatar",
            Message = "Invalid value."
        });
        flag = false;
    }
}
if (flag)
                {
                    successData["avatar"] = data["avatar"];
                }
            }
            {
                flag = true;bool flag = true;

if (!data.TryGetValue("sessionToken", out var rawValue) || rawValue == null)
{
    result.Errors.Add(new ValidationError
    {
        Path = "sessionToken",
        Message = "Invalid value."
    });
    flag = false;
}
else if (rawValue is not string value)
{
    result.Errors.Add(new ValidationError
    {
        Path = "sessionToken",
        Message = "Invalid value."
    });
    flag = false;
}
else if (!Base64Validator.IsStrictBase64(value, true)) {
    result.Errors.Add(new ValidationError {
        Path = "sessionToken",
        Message = "Invalid value."
    });
    flag = false;
} else {
    byte[] decodedBytes = Convert.FromBase64String(value.Replace('-', '+').Replace('_', '/'));
    if (decodedBytes.Length < 32) {
        result.Errors.Add(new ValidationError {
            Path = "sessionToken",
            Message = "Invalid value."
        });
        flag = false;
    }
    if (decodedBytes.Length > 64) {
        result.Errors.Add(new ValidationError {
            Path = "sessionToken",
            Message = "Invalid value."
        });
        flag = false;
    }
    value = value.Trim();
    data["sessionToken"] = value;
}
if (flag)
                {
                    successData["sessionToken"] = data["sessionToken"];
                }
            }
            {
                flag = true;bool flag = true;

if (!data.TryGetValue("tags", out var rawValue) || rawValue == null)
{
    result.Errors.Add(new ValidationError
    {
        Path = "tags",
        Message = "Invalid value."
    });
    flag = false;
}
else if (rawValue is not System.Collections.IEnumerable enumerable || rawValue is string)
{
    result.Errors.Add(new ValidationError
    {
        Path = "tags",
        Message = "Invalid value."
    });
    flag = false;
}
else
{
    var list = enumerable.Cast<object?>().ToList();

    for (int i = 0; i < list.Count; i++)
    {
        var previousFlag = flag;
        flag = true;

        var itemPath = "tags" + "[" + i + "]";
        data[itemPath] = list[i];bool flag = true;

if (!data.TryGetValue(itemPath, out var rawValue) || rawValue == null)
{
    result.Errors.Add(new ValidationError
    {
        Path = itemPath,
        Message = "Invalid value."
    });
    flag = false;
}
else if (rawValue is not string value)
{
    result.Errors.Add(new ValidationError
    {
        Path = itemPath,
        Message = "Invalid value."
    });
    flag = false;
}
else if (!Base64Validator.IsStrictBase64(value, false)) {
    result.Errors.Add(new ValidationError {
        Path = itemPath,
        Message = "Invalid value."
    });
    flag = false;
} else {
}


        list[i] = data[itemPath];
        data.Remove(itemPath);

        var itemFlag = flag;
        flag = previousFlag && itemFlag;}

    data["tags"] = list;
}if (flag)
                {
                    successData["tags"] = data["tags"];
                }
            }

            if (result.Errors.Count == 0)
            {
                result = new ValidationResult
                {
                    Success = true,
                    Errors = new List<ValidationError>(),
                    Data = successData
                };
            }

            return result;
        }
    }
}